### PR TITLE
GraphQl-673: isEmailAvailable test coverage

### PR DESCRIPTION
**Pull request by moloughlin**
#673

- Add email validation on the isEmailAvailable query
- Add test coverage for negative cases of isEmailAvailable query

### CODE REVIEW

**Review by VitaliyBoyko — APPROVED** [review]

**Comment by naydav** [comment]
Hello @moloughlin,
Could you resolve conflict with mainline? Thank you.

**Comment by moloughlin** [comment]
@naydav - done, cheers!

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/IsEmailAvailableTest.php [hunk 1]
```diff
@@ -47,4 +47,55 @@ public function testEmailAvailable()
         self::assertArrayHasKey('is_email_available', $response['isEmailAvailable']);
         self::assertTrue($response['isEmailAvailable']['is_email_available']);
     }
+
+    /**
+     * @expectedException \Exception
+     * @expectedExceptionMessage GraphQL response contains errors: Email should be specified
```
Comment (lenaorobei) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/IsEmailAvailableTest.php]: Looks like wrong error message expected.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please consider fixing code style issues https://testing-service.magento-community.engineering/reports/magento/graphql-ce/pull/706/290768a3-48da-470f-9213-92802f572f0a/16025/Statics/allure-report-ce/index.html#suites/51d3ab4d5d77a0bc6a7ea344e9e7e6be

And GraphQl API tests - https://github.com/magento/graphql-ce/pull/706/checks?check_run_id=141859163


Please let me know if you need any help resolving those issues.

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/IsEmailAvailableTest.php (hunk 1 above)
Comment (moloughlin) [reply dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/IsEmailAvailableTest.php]: Thanks for that, I hadn't noticed the message change when merging in the latest 2.3-develop :+1: 

I've addressed this point, and the code style issues

**Comment by lenaorobei** [comment]
@moloughlin could you please check failed builds? Please, let me know if you need any help here.
Thanks.

**Comment by lenaorobei** [comment]
@magento run all tests

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5233](https://jira.corp.magento.com/browse/ENGCOM-5233) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@moloughlin thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.
